test/mint.js

```diff
@@ -447,4 +447,55 @@ describe('PodCastNFT', function () {
       expect((await contract.userAttribute(alice.address)).point).to.eq(0)
     })
   })
+
+  describe('transfer', async () => {
+    it('revert without nft', async () => {
+      await expect(
+        contract.setRoles(alice.address, ['MEMBER', 'ADMIN'])
+      ).to.be.revertedWith('MUST BE HOLDER')
+    })
+
+    it('revert without token ID', async () => {
+      const mintTx = await contract.mint(
+        'https://example.com/podcast.png',
+        ['Podcast Contributor'],
+        alice.address
+      )
+      await mintTx.wait()
+
+      await expect(
+        contract.connect(alice).transfer(bob.address, 2, 1)
+      ).to.be.revertedWith('ERC721Metadata: URI set of nonexistent token')
+    })
+
+    it('can transfer', async () => {
+      const roles = ['Podcast Contributor', 'MEMBER']
+      const mintTx = await contract.mint(
+        'https://example.com/podcast.png',
+        roles,
+        alice.address
+      )
+      await mintTx.wait()
+
+      const point = 100
+      const giveAwayPointTx = await contract.giveAwayPoint(alice.address, point)
+      await giveAwayPointTx.wait()
+
+      const transferTx = await contract
+        .connect(alice)
+        .transfer(bob.address, 1, 1)
+      await transferTx.wait()
+
+      expect(await contract.balanceOf(alice.address)).to.eq(0)
+      expect(contract.roles(alice.address, 0)).to.be.reverted
+      expect(contract.userAttribute(alice.address)).to.be.reverted
+      expect(await contract.balanceOf(bob.address)).to.eq(1)
+      expect(await contract.roles(bob.address, 0)).to.eq(roles[0])
+      expect(await contract.roles(bob.address, 1)).to.eq(roles[1])
+      expect((await contract.userAttribute(bob.address)).point).to.eq(point)
+      expect((await contract.userAttribute(bob.address)).claimableToken).to.eq(
```

Comment:
[IMO] I noticed while implementing the test that there is no need to increase the claimableToken in the giveAwayPoint?

Reply:
@geeknees can you fix this ?
https://github.com/henkaku-center/podcast-nft/pull/52/files/bd8d9c8e94ee6dd1e675f2137972777da7238233#diff-091a64f3ece8473484ef0f9f24718ed5e3f9e669991fa082bbbc30719df20459